pkg/network/node/node.go

```diff
@@ -296,71 +294,19 @@ func (node *OsdnNode) validateMTU() error {
 			continue
 		}
 
-		newmtu := link.Attrs().MTU
-		if newmtu > 0 && newmtu < mtu {
-			mtu = newmtu
+		mtu := link.Attrs().MTU
+		if mtu > 0 && mtu < interfaceMTU {
+			interfaceMTU = mtu
 		}
 	}
-	if mtu > maxMTU {
-		return fmt.Errorf("unable to determine MTU while performing validation")
+	if interfaceMTU > maxMTU {
+		klog.Errorf("Unable to determine MTU while performing validation. (Assuming MTU is valid.)")
+		return nil
 	}
 
-	needsTaint := mtu < int(node.mtu)+50
-	const MTUTaintKey string = "network.openshift.io/mtu-too-small"
-	mtuTooSmallTaint := &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"}
-	nodeObj, err := node.kClient.CoreV1().Nodes().Get(context.TODO(), node.hostName, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
-	}
-	tainted := taints.TaintExists(nodeObj.Spec.Taints, mtuTooSmallTaint)
-	if needsTaint != tainted {
-		resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			nodeObj, err = node.kClient.CoreV1().Nodes().Get(context.TODO(), node.hostName, metav1.GetOptions{})
-			if err != nil {
-				return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
-			}
-			nodeObjCopy := nodeObj.DeepCopy()
-			var nodeWithTaintMods *corev1.Node
-
-			if needsTaint && !tainted {
-				klog.V(2).Infof("Default interface MTU is less than VXLAN overhead, tainting node...")
-				nodeWithTaintMods, _, err = taints.AddOrUpdateTaint(nodeObjCopy, mtuTooSmallTaint)
-				if err != nil {
-					return fmt.Errorf("could not taint the node with key %s: %v", MTUTaintKey, err)
-				}
-			} else if !needsTaint && tainted {
-				klog.V(2).Infof("Node has too small MTU taint but default interface MTU is big enough, untainting node...")
-				nodeWithTaintMods, _, err = taints.RemoveTaint(nodeObjCopy, mtuTooSmallTaint)
-				if err != nil {
-					return fmt.Errorf("could not untaint the node with key %s: %v", MTUTaintKey, err)
-				}
-			}
-
-			nodeObjJson, err := json.Marshal(nodeObj)
-			if err != nil {
-				return fmt.Errorf("could not marshal old Node object: %v", err)
-			}
-
-			newNodeObjJson, err := json.Marshal(nodeWithTaintMods)
-			if err != nil {
-				return fmt.Errorf("could not marshal new Node object: %v", err)
-			}
-
-			patchBytes, err := strategicpatch.CreateTwoWayMergePatch(nodeObjJson, newNodeObjJson, corev1.Node{})
-			if err != nil {
-				return fmt.Errorf("could not create patch for object: %v", err)
-			}
-
-			_, err = node.kClient.CoreV1().Nodes().Patch(context.TODO(), node.hostName, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
-
-			return err
-		})
-
-		if resultErr != nil {
-			return fmt.Errorf("could not update node taints after many retries: %v", resultErr)
-		}
+	if interfaceMTU < int(node.overlayMTU)+50 {
```

Comment:
ok this is the fix u described in this PR, why did u change the error code in the above codes to nil ?

Reply:
`validateMTU` has been changed so that it only returns an error if the MTU is definitely wrong (in which case its caller will cause the program to exit). In all of the cases where it runs into a problem trying to figure out if the MTU is right or wrong, it now just logs an error message and then returns `nil`, assuming that _probably_ the MTU is right and it should keep going.

Reply:
/lgtm